src/octoprint/server/api/timelapse.py

```diff
@@ -113,6 +116,21 @@ def setTimelapseConfig():
 				else:
 					return make_response("Invalid value for interval: %d" % interval)
 
+		if "retractionZHop" in request.values:
+                        config["options"] = {
+                                "retractionZHop": 0
+                        }
+
+                        try:
+                                retractionZHop = float(request.values["retractionZHop"])
+                        except ValueError:
+                                return make_response("Invalid value for retraction Z-Hop: %r" % request.values["retractionZHop"])
+                        else:
+                                if retractionZHop > 0:
```

Comment:
This doesn't allow retractionZHop to be 0, but that is the default lower down.  The effect is that if you switch to "Timed" and hit "Save Config" and then switch back to "On Z Change" it won't let you "Save Config" any more unless you set the retraction Z-hop to something non-zero.  Is there a problem letting it be zero?  If so, should we fix that problem or set the default to a small non-zero number (0.01mm)?